src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SearchParameterQueryGenerator.cs

```diff
@@ -91,6 +108,21 @@ public override SearchParameterQueryGeneratorContext VisitMultiary(MultiaryExpre
             return context;
         }
 
+        protected static SearchParameterQueryGenerator GetSearchParameterQueryGeneratorIfResourceColumnSearchParameter(SearchParameterExpressionBase searchParameter)
+        {
+            switch (searchParameter.Parameter.Name)
```

Comment:
>searchParameter.Parameter.Name [](start = 20, length = 30)

This look very similar to `ColumnLocation` in `SearchParameterInfoExtensions`.
Is lastUpdated got omitted here because it has special handling case somewhere else?

Reply:
At this point, LastUpdated should have been replaced by ResourceSurrogateId. I could add a Debug.Assert or throw if we encounter LastUpdated.